File: hailiao-api/src/main/java/com/hailiao/api/dto/GroupJoinRequestDTO.java
package com.hailiao.api.dto;

import java.util.Date;

public class GroupJoinRequestDTO {
    private Long id;
    private Long groupId;
    private Long userId;
    private String message;
    private Integer status;
    private Long handledBy;
    private Date handledAt;
    private Date createdAt;
    private UserDTO userInfo;
    private GroupDTO groupInfo;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getGroupId() { return groupId; }
    public void setGroupId(Long groupId) { this.groupId = groupId; }

    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }

    public String getMessage() { return message; }
    public void setMessage(String message) { this.message = message; }

    public Integer getStatus() { return status; }
    public void setStatus(Integer status) { this.status = status; }

    public Long getHandledBy() { return handledBy; }
    public void setHandledBy(Long handledBy) { this.handledBy = handledBy; }

    public Date getHandledAt() { return handledAt; }
    public void setHandledAt(Date handledAt) { this.handledAt = handledAt; }

    public Date getCreatedAt() { return createdAt; }
    public void setCreatedAt(Date createdAt) { this.createdAt = createdAt; }

    public UserDTO getUserInfo() { return userInfo; }
    public void setUserInfo(UserDTO userInfo) { this.userInfo = userInfo; }

    public GroupDTO getGroupInfo() { return groupInfo; }
    public void setGroupInfo(GroupDTO groupInfo) { this.groupInfo = groupInfo; }
}

File: hailiao_flutter/lib/models/group_join_request_dto.dart
import 'user_dto.dart';
import 'group_dto.dart';

class GroupJoinRequestDTO {
  final int? id;
  final int? groupId;
  final int? userId;
  final String? message;
  final int? status;
  final int? handledBy;
  final String? handledAt;
  final String? createdAt;
  final UserDTO? userInfo;
  final GroupDTO? groupInfo;

  GroupJoinRequestDTO({
    this.id,
    this.groupId,
    this.userId,
    this.message,
    this.status,
    this.handledBy,
    this.handledAt,
    this.createdAt,
    this.userInfo,
    this.groupInfo,
  });

  factory GroupJoinRequestDTO.fromJson(Map<String, dynamic> json) {
    return GroupJoinRequestDTO(
      id: (json['id'] as num?)?.toInt(),
      groupId: (json['groupId'] as num?)?.toInt(),
      userId: (json['userId'] as num?)?.toInt(),
      message: json['message'] as String?,
      status: (json['status'] as num?)?.toInt(),
      handledBy: (json['handledBy'] as num?)?.toInt(),
      handledAt: json['handledAt'] as String?,
      createdAt: json['createdAt'] as String?,
      userInfo: json['userInfo'] is Map<String, dynamic>
          ? UserDTO.fromJson(json['userInfo'] as Map<String, dynamic>)
          : null,
      groupInfo: json['groupInfo'] is Map<String, dynamic>
          ? GroupDTO.fromJson(json['groupInfo'] as Map<String, dynamic>)
          : null,
    );
  }
}
